lib/archethic_web/controllers/api/origin_key_controller.ex

```diff
@@ -3,35 +3,37 @@ defmodule ArchethicWeb.API.OriginKeyController do
 
   alias Archethic.Crypto
   alias Archethic.SharedSecrets
-
-  alias Archethic.TransactionChain
-  alias Archethic.TransactionChain.TransactionData.Ownership
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.TransactionData
 
   def origin_key(conn, params) do
-    with %{"origin_public_key" => origin_public_key} <- params,
+    with %{"origin_public_key" => origin_public_key, "certificate" => certificate} <- params,
          {:ok, origin_public_key} <- Base.decode16(origin_public_key, case: :mixed),
          true <- Crypto.valid_public_key?(origin_public_key),
-         <<_curve_id::8, origin_id::8, _rest::binary>> <- origin_public_key,
-         {first_origin_family_public_key, _} <-
-           SharedSecrets.get_origin_family_from_origin_id(origin_id)
-           |> SharedSecrets.get_origin_family_seed()
-           |> Crypto.derive_keypair(0),
-         {:ok, tx} <-
-           Crypto.derive_address(first_origin_family_public_key)
-           |> TransactionChain.get_last_transaction(data: [:ownerships]),
-         ownership when ownership != nil <-
-           Enum.find(tx.data.ownerships, fn ownership ->
-             Ownership.authorized_public_key?(ownership, origin_public_key)
-           end) do
-      res = %{
-        encrypted_origin_private_keys: Base.encode16(ownership.secret),
-        encrypted_secret_key:
-          Ownership.get_encrypted_key(ownership, origin_public_key) |> Base.encode16()
-      }
+         true <- Crypto.get_key_certificate(origin_public_key) == certificate,
+         address <- Crypto.derive_address(origin_public_key),
+         {:ok, last_address} <-
+           Archethic.get_last_transaction_address(address),
+         {:ok, last_index} <- Archethic.get_transaction_chain_length(last_address) do
+      <<_curve_id::8, origin_id::8, _rest::binary>> = origin_public_key
+      origin_family = Crypto.key_origin(origin_id)
+      signing_seed = SharedSecrets.get_origin_family_seed(origin_family)
+
+      tx =
+        Transaction.new(
+          :origin,
+          %TransactionData{
+            content: <<origin_public_key::binary>>
```

Comment:
We should add the certificate in the content as it should be validated by the network.
Also the check of the certificate is not correct as we have not the certificate in the node config but only the root CA.
So the best is to send the transaction and let the validation nodes makes the check like in there https://github.com/archethic-foundation/archethic-node/blob/76cefaceccc1cf0291622be10e473cd12d148475/lib/archethic/mining/pending_transaction_validation.ex#L132

Reply:
This means we have to concat certificate along with public key right?
Do we have any standard byte size for publickey to set and retrieve second part as a content?

Reply:
Yeah, updated the validation part for certificate adding inside the content. I have added new private function to match transaction type `origin`.